src/io.rs

```diff
@@ -216,7 +216,7 @@ impl TrainInfo {
 
     /// Get the end datetime.
     pub fn end_datetime(&self) -> Option<&str> {
-        self.end_datetime.as_ref().map(|s| s.as_str())
+        self.end_datetime.as_deref()
```

Comment:
That's handy, new feature?

Reply:
Yeah, I found out about this because `cargo clippy` pointed out this handy new feature.